pkg/controller/operatingsystemconfig/actuator.go

```diff
@@ -9,23 +9,35 @@ import (
 	_ "embed"
 	"fmt"
 	"path/filepath"
+	"strings"
+	"text/template"
 
 	"github.com/gardener/gardener/extensions/pkg/controller/operatingsystemconfig"
 	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
 	"github.com/go-logr/logr"
 	"k8s.io/utils/ptr"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	configv1alpha1 "github.com/gardener/gardener-extension-os-coreos/pkg/controller/config/v1alpha1"
 )
 
 type actuator struct {
-	client client.Client
+	client          client.Client
+	extensionConfig Config
+}
+
+// Config contains configuration for the extension service.
+type Config struct {
+	// Embed the entire Extension config here for direct access in the controller.
+	*configv1alpha1.ExtensionConfig
```

Comment:
Could we directly use the ExtensionConfig until there's a reason to wrap it?

Reply:
We could but I don't see why this would so horribly.

I expect this to gain more options in the future.